src/components/periodic-report/handlers/sync-periodic-report-to-project.handler.ts

```diff
@@ -0,0 +1,115 @@
+import { ProjectUpdatedEvent } from '../../project/events';
+import { CreatePeriodicReport } from '../dto';
+import { ReportType } from '../dto/type.enum';
+import { PeriodicReportService } from '../periodic-report.service';
+
+type SubscribedEvent = ProjectUpdatedEvent;
+
+@EventsHandler(ProjectUpdatedEvent)
+export class SyncPeriodicReportToProject
+  implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly periodicReports: PeriodicReportService,
+    @Logger('periodicReport:sync-project') private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Project mutation, syncing periodic reports', {
+      ...event,
+      event: event.constructor.name,
+    });
+
+    const project = event.updated;
+    const previous = event.previous;
+
+    const isDateRangeChanged =
+      project.mouStart !== previous.mouStart ||
+      project.mouEnd !== previous.mouEnd;
+
+    if (isDateRangeChanged) {
+      await this.removeOldReports(project, event.session);
+      await this.syncRecords(project, event.session);
+    }
+  }
+
+  private async removeOldReports(project: Project, session: Session) {
+    const reports = await this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: project.id }),
+        relation('out', '', 'report', { active: true }),
+        node('report', 'PeriodicReport'),
+      ])
+      .optionalMatch([
+        node('report'),
+        relation('out', 'rel', 'reportFile', { active: true }),
+        node('file', 'File'),
+      ])
+      .with('report, rel')
+      .where({ rel: isNull() })
+      .return('report.id as reportId')
+      .asResult<{ reportId: string }>()
+      .run();
+
+    await Promise.all(
+      reports.map((report) =>
+        this.periodicReports.delete(report.reportId, session)
+      )
+    );
+  }
+
+  private async syncRecords(project: Project, session: Session) {
+    await this.addReport(
+      project.id,
+      {
+        start: project.mouStart.value!,
+        end: project.mouEnd.value!,
+        type: ReportType.Financial,
+      },
+      session
+    );
+
+    await this.addReport(
+      project.id,
+      {
+        start: project.mouStart.value!,
+        end: project.mouEnd.value!,
+        type: ReportType.Narrative,
+      },
+      session
+    );
+  }
+
+  private async addReport(
+    projectId: string,
+    input: CreatePeriodicReport,
+    session: Session
+  ) {
+    const report = await this.periodicReports.create(input, session);
+    await this.db
+      .query()
+      .match(node('project', 'Project', { id: projectId }))
+      .match(node('periodicReport', 'PeriodicReport', { id: report.id }))
+      .create([
+        node('project'),
+        relation('out', '', 'report', {
+          active: true,
+          createdAt: DateTime.local(),
+        }),
+        node('periodicReport'),
+      ])
+      .run();
```

Comment:
A report should always be attached to a project. This logic should be encapsulated in the service. I understand this might need type changes, feel free to make those as well for creation.